java-grupa8/Dealership/src/main/java/ro/scii/java8/Customer.java

```diff
@@ -0,0 +1,395 @@
+package ro.scii.java8;
+
+
+import java.io.*;
+import java.util.*;
+
+/**
+ * @author x-RauL-x
+ * This class represents the available actions of the customer towards the dealership.
+ */
+public class Customer {
+
+    private String customerName;
+    private int customerId;
+    private int budget;
+    private int option;
+    private GreenBonusProgram gb = new GreenBonusProgram();
+    private Dealership checkStock = new Dealership();
+    private Dealership requestGBP = new Dealership();
+    private Scanner s = new Scanner(System.in);
+    private int checkProgram;
+
+    public Customer(String customerName, int budget) {
+        this.customerName = customerName;
+        this.budget = budget;
+    }
+
+    public int getBudget() {
+        return budget;
+    }
+
+
+    /**
+     * This method displays all the vehicles from a list
+     *
+     * @param list the list that is displayed
+     * @throws IOException
+     * @throws ClassNotFoundException
+     */
+    public void viewAllVehiclesList(List<ElectricVehicle> list) throws IOException, ClassNotFoundException {
+
+        for (ElectricVehicle v : list) {
+            System.out.println(v.getManufacturer() + "\n Model: " + v.getVehicleModel() + "\n Power: " + v.getMotorPower() + " kW " +
+                    "\t Battery: " + v.getBatteryCapacity() + " kWh" + "\t Range: " + v.getRange() + " km " + "\t Fast Charging: " + v.isHasFastCharging() +
+                    "\t New: " + v.isNew() + "\t Price: " + v.getPrice() + " Euro" + "\t Stock No.: " + v.getStockNr() + "\n");
+        }
+        sortVehicles(list);
+    }
+
+
+    /**
+     * This method sorts a vehicle list based on parameters chosen by the customer from the keyboard
+     *
+     * @param list the list that is sorted
+     * @throws IOException
+     * @throws ClassNotFoundException
+     */
+    public void sortVehicles(List<ElectricVehicle> list) throws IOException, ClassNotFoundException {
```

Comment:
This method does not belong to the customer. The customer might be calling it on a dealership.